src/components/Register/Register.jsx

```diff
@@ -0,0 +1,179 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { signupUser } from '../../store/modules/auth';
+import {
+  validateInput,
+  checkAllEmptyInputs,
+  fieldChecker,
+} from '../../utils/formValidator';
+
+// components
+import Button from '../presentationals/Button/Button';
+import Image from '../presentationals/Image/Image';
+import Loader from '../presentationals/Loader/Loader';
+import '../../components/presentationals/AuthenticationCard/AuthenticationCard.scss';
+
+// styles
+import '../../components/presentationals/Form.scss';
+
+// images
+import GoogleIcon from '../../assets/images/google-icon.svg';
+import FacebookIcon from '../../assets/images/facebook-icon.svg';
+import TwitterIcon from '../../assets/images/twitter-icon.svg';
+
+class Register extends Component {
+  state = {
+    userCredentials: {
+      email: '',
+      password: '',
+      username: '',
+    },
+    validationErrors: [],
+    emptyFields: [],
+  };
+
+  userInputHandler = event => {
+    const { emptyFields } = this.state;
+    const currentEmptyFields = fieldChecker(event, emptyFields);
+    this.setState({
+      userCredentials: {
+        ...this.state.userCredentials,
+        [event.target.name]: event.target.value,
+      },
+      emptyFields: currentEmptyFields,
+    });
+  };
+
+  formValidator = event => {
+    event.preventDefault();
+    const { userCredentials } = this.state;
+    const emptyFields = checkAllEmptyInputs(userCredentials);
+    if (emptyFields.length) {
+      this.setState({
+        emptyFields,
+      });
+      return false;
+    }
+    this.setState({ emptyFields: [] });
+    const validationErrors = validateInput(userCredentials);
+    this.setState({
+      validationErrors,
+    });
+    if (!validationErrors.length) {
+      this.setState({ validationErrors: '' });
+      return true;
+    }
+    return false;
+  };
+
+  submitForm = event => {
+    event.preventDefault();
+    this.props.signupUser({
+      ...this.state.userCredentials,
+    });
+  };
+
+  render() {
+    const { auth } = this.props;
+    const { emptyFields } = this.state;
+    const usernameClasses = ['form-control'];
```

Comment:
Why not apply these classes directly to the component.

Reply:
I have refactored to reflect this now.
```<input
              className={`form-control ${
                emptyFields.includes('username') ? 'error' : ''
              }`}
              placeholder="Username"
              onChange={this.userInputHandler}
              value={this.state.userCredentials.username}
              name="username"
            />
```